diff --git a/node_modules/sequelize/types/model.d.ts b/node_modules/sequelize/types/model.d.ts
index ccdb79d..47b42b0 100644
--- a/node_modules/sequelize/types/model.d.ts
+++ b/node_modules/sequelize/types/model.d.ts
@@ -55,14 +55,14 @@ export interface Filterable<TAttributes = any> {
   where?: WhereOptions<TAttributes>;
 }
 
-export interface Projectable {
+export interface Projectable<TAttributes> {
   /**
    * A list of the attributes that you want to select. To rename an attribute, you can pass an array, with
    * two elements - the first is the name of the attribute in the DB (or some kind of expression such as
    * `Sequelize.literal`, `Sequelize.fn` and so on), and the second is the name you want the attribute to
    * have in the returned instance
    */
-  attributes?: FindAttributeOptions;
+  attributes?: FindAttributeOptions<TAttributes>;
 }
 
 export interface Paranoid {
@@ -739,15 +739,15 @@ export type Order = Fn | Col | Literal | OrderItem[];
  */
 export type ProjectionAlias = readonly [string | Literal | Fn | Col, string];
 
-export type FindAttributeOptions =
-  | (string | ProjectionAlias)[]
+export type FindAttributeOptions<TAttributes> =
+  | (keyof TAttributes | ProjectionAlias)[]
   | {
-    exclude: string[];
-    include?: (string | ProjectionAlias)[];
+    exclude: (keyof TAttributes)[] | string[];
+    include?: (keyof TAttributes | ProjectionAlias)[];
   }
   | {
-    exclude?: string[];
-    include: (string | ProjectionAlias)[];
+    exclude?: (keyof TAttributes)[] | string[];
+    include: (keyof TAttributes | ProjectionAlias)[];
   };
 
 export interface IndexHint {
@@ -768,7 +768,7 @@ export interface IndexHintable {
  * A hash of options to describe the scope of the search
  */
 export interface FindOptions<TAttributes = any>
-  extends QueryOptions, Filterable<TAttributes>, Projectable, Paranoid, IndexHintable
+  extends QueryOptions, Filterable<TAttributes>, Projectable<TAttributes>, Paranoid, IndexHintable
 {
   /**
    * A list of associations to eagerly load using a left join (a single association is also supported). Supported is either
